### ISSUE TITLE
Modules

### ISSUE THREAD

**Issue by Gustorn** [tomassedovic/tcod-rs#136]
I think it would be beneficial to split the project into modules for two reasons:

* if there'll be safe safe wrappers around everything in `libtcod` then a single `.rs` file will be really hard to handle
* it would probably make the life of new contributors  easier
* the module structure is already pretty well defined on the main project page, an example split:

| Name        |  Name       | Name    |
|----------------|---------------|------------|
| chars         | input          | system |
| colors        | map           |
| console     | pathfinding |

If we still want to have everything available from the top module we can just reexport the necessary `structs`, `traits` and `functions`.

**Comment by tomassedovic** [comment]
Since modules are orthogonal to files in Rust, are you talking about one of the two or both?

We do have some modules already (system, colors, mouse) and I'm definitely okay to add new ones and move the stuff that's currently top-level in them.

As for splitting the main file, I agree that if we have everything, it'll probably be too big. Personally, the size of `src/lib.rs` doesn't bother me that much yet, but if it's too much for other contributors, let's do it.

I would want to keep some re-exports but mostly to provide simple ergonomics. For the record, the code that's currently in the root module is not there because we wanted it so, but because that's where we happened to put it.

The module names sound right to me, too. We can also look at libtcod's documentation -- they don't really split stuff into modules, but the docs are organised based on functionality that we can reuse.

**Comment by Gustorn** [comment]
Yeah sorry, I meant splitting the modules into separate files, I saw that they are in different modules already. I cannot speak for anyone else, but I like having multiple (logically split) source files when working on someone elses code. I can manage really long files that I wrote myself (or someone else if I take my time understanding it).

But for new contributors (especially ones that wants to fix bugs), only having to look through a small portion of the code is probably easier.
